src/AIAPI/MainGame/GameExtensions.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+// using ActionGame.Chara;
+using AIProject;
+using AIProject.SaveData;
+using AIChara;
+
+namespace KKAPI.MainGame
+{
+    /// <summary>
+    /// Extensions useful in the main game
+    /// </summary>
+    public static class GameExtensions
+    {
+
+        /// <summary>
+        /// Get the persisting heroine object that describes this character.
+        /// Returns null if the heroine could not be found. Works only in the main game.
+        /// </summary>
+        public static Actor GetHeroine(this ChaControl chaControl)
+        {
+            if (chaControl == null) throw new ArgumentNullException(nameof(chaControl));
+
+            if (!Manager.Map.IsInstance()) return null;
+            return Manager.Map.Instance.Actors.Find(heroine => heroine.ChaControl == chaControl);
+        }
+
+        /// <summary>
+        /// Get the persisting heroine object that describes this character.
+        /// Returns null if the heroine could not be found. Works only in the main game.
+        /// </summary>
+        public static Actor GetHeroine(this ChaFileControl chaFile)
+        {
+            if (chaFile == null) throw new ArgumentNullException(nameof(chaFile));
+
+            if (!Manager.Map.IsInstance()) return null;
+            return Manager.Map.Instance.Actors.Find(heroine => heroine.ChaControl?.chaFile == chaFile);
+        }
+
+        /// <summary>
+        /// Get the NPC that represents this heroine in the game. Works only in the main game.
+        /// If the heroine has not been spawned into the game it returns null.
+        /// </summary>
+        public static Actor GetNPC(this AgentActor heroine)
+        {
+            if (heroine == null) throw new ArgumentNullException(nameof(heroine));
+
+            if (heroine.transform == null) return null;
+            return heroine.transform.GetComponent<Actor>();
+        }
+
+        /// <summary>
+        /// Get ChaFiles that are related to this heroine. Warning: It might not return some copies.
+        /// </summary>
+        public static IEnumerable<ChaFileControl> GetRelatedChaFiles(this AgentActor heroine)
```

Comment:
Is AgentActor the correct type to be used here to fetch the ChaFileControl?  Or should it just be the Actor?